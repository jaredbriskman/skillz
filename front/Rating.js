import React, { Component } from 'react';

export const Rating = ({ rating, icon }) =>
    <div className="compact my-rating">
        {Array.apply(null, Array(rating || 0)).map((_, i) =>
            <i key={i} className={"icon small " + (icon || 'star')} />)}
    </div>

export class EditRating extends Component {
    constructor(props) {
        super(props);
        this.state = { rating: props.rating };
        this.onRating = this.onRating.bind(this);
    }
    onRating(rating) {
        this.props.onRating(rating);
        this.setState({ rating });
    }
    render() {
        let { props, state, onRating } = this;
        const icon = props.icon || 'star';
        const rating = state.rating || 0;
        return (
            <div className="my-rating">
                <i className={"small circle icon" + (rating ? " thin" : "")}
                    onClick={() => onRating(null)} />
                {rating > 0 && Array.apply(null, Array(rating)).map((_, i) =>
                    <i key={i} className={"icon small " + icon}
                        onClick={() => onRating(i + 1)} />)}
                {rating < 5 && Array.apply(null, Array(5 - rating)).map((_, i) =>
                    <i key={i + rating} className={"icon small empty " + icon}
                        onClick={() => onRating(i + 1 + rating)} />)}
            </div>)
    }
}
