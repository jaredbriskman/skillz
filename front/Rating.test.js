import { EditRating, Rating } from './Rating.js'

import React from 'react'
import renderer from 'react-test-renderer';

describe('Rating', () => {
    test('makes stars', () => {
        const component = renderer.create(
            <Rating rank="2" />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})


describe('EditRating', () => {
    test('makes stars', () => {
        const component = renderer.create(
            <EditRating rank="2" />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})
