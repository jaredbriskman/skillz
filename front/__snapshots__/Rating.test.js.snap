// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRating makes stars 1`] = `
<div
  className="my-rating"
>
  <i
    className="small circle icon"
    onClick={[Function]}
  />
  <i
    className="icon small empty star"
    onClick={[Function]}
  />
  <i
    className="icon small empty star"
    onClick={[Function]}
  />
  <i
    className="icon small empty star"
    onClick={[Function]}
  />
  <i
    className="icon small empty star"
    onClick={[Function]}
  />
  <i
    className="icon small empty star"
    onClick={[Function]}
  />
</div>
`;

exports[`Rating makes stars 1`] = `
<div
  className="compact my-rating"
>
  <i
    className="icon small star"
  />
</div>
`;
